internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,424 @@
+
+		iamRequest := &organizationiammodel.VmwareTanzuManageV1alpha1OrganizationPatchOrganizationIAMPolicyRequest{
+			FullName:         fn,
+			BindingDeltaList: blData,
+		}
+
+		iamResponse, err := config.TMCConnection.OrganizationIAMResourceService.ManageV1alpha1OrganizationIAMPolicyPatch(iamRequest)
+		if err != nil {
+			return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for organization"))
+		}
+
+		d.SetId(iamResponse.Policy.Meta.UID)
+	case clusterGroup:
+		fn, _ := fnData.(*clustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFullName)
+
+		iamRequest := &clustergroupiammodel.VmwareTanzuManageV1alpha1ClustergroupPatchClusterGroupIAMPolicyRequest{
+			FullName:         fn,
+			BindingDeltaList: blData,
+		}
+
+		iamResponse, err := config.TMCConnection.ClusterGroupIAMResourceService.ManageV1alpha1ClusterGroupIAMPolicyPatch(iamRequest)
+		if err != nil {
+			return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for cluster group"))
+		}
+
+		d.SetId(iamResponse.Policy.Meta.UID)
+	case cluster:
+		fn, _ := fnData.(*clustermodel.VmwareTanzuManageV1alpha1ClusterFullName)
+
+		iamRequest := &clusteriammodel.VmwareTanzuManageV1alpha1ClusterPatchClusterIAMPolicyRequest{
+			FullName:         fn,
+			BindingDeltaList: blData,
+		}
+
+		iamResponse, err := config.TMCConnection.ClusterIAMResourceService.ManageV1alpha1ClusterIAMPolicyPatch(iamRequest)
+		if err != nil {
+			return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for cluster"))
+		}
+
+		d.SetId(iamResponse.Policy.Meta.UID)
+	case workspace:
+		fn, _ := fnData.(*workspacemodel.VmwareTanzuManageV1alpha1WorkspaceFullName)
+
+		iamRequest := &workspaceiammodel.VmwareTanzuManageV1alpha1WorkspacePatchWorkspaceIAMPolicyRequest{
+			FullName:         fn,
+			BindingDeltaList: blData,
+		}
+
+		iamResponse, err := config.TMCConnection.WorkspaceIAMResourceService.ManageV1alpha1WorkspaceIAMPolicyPatch(iamRequest)
+		if err != nil {
+			return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for workspace"))
+		}
+
+		d.SetId(iamResponse.Policy.Meta.UID)
+	case namespace:
+		fn, _ := fnData.(*namespacemodel.VmwareTanzuManageV1alpha1ClusterNamespaceFullName)
+
+		iamRequest := &namespaceiammodel.VmwareTanzuManageV1alpha1ClusterNamespacePatchNamespaceIAMPolicyRequest{
+			FullName:         fn,
+			BindingDeltaList: blData,
+		}
+
+		iamResponse, err := config.TMCConnection.NamespaceIAMResourceService.ManageV1alpha1ClusterNamespaceIAMPolicyPatch(iamRequest)
+		if err != nil {
+			return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for namespace"))
+		}
+
+		d.SetId(iamResponse.Policy.Meta.UID)
+	default:
+		return diag.FromErr(errors.Wrap(err, "unable to create Role Binding, invalid scope defined"))
+	}
+
+	return dataSourceIAMPolicyRead(ctx, d, m)
+}
+
+func resourceIAMPolicyDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
Is this required?

Reply:
Yes, for `terraform destroy` to work

Reply:
I mean, this was not implemented. Anyways you can add this incrementally.